### ISSUE TITLE
Remove .data from test/test_autograd.py for fixathon

### ISSUE THREAD

**Issue by gchanan** [pytorch/pytorch#33630]
Even though it is not documented, many users still use it. And it leads to many bugs in user code.
So we should remove it completely to prevent this.

To do this, do one of:
1) [preferred] remove `.data`
2) [if possible] otherwise, replace with either `.detach()` or wrap in a `torch.no_grad` block
3) [otherwise] leave as is.

For 2), generally use whichever is nicer.  `torch.no_grad()` is often clearer but is extra characters and is often done on another line.  One place where you may have to use `.detach()` is in the return of a function, i.e.:
`return x.detach()` returns a tensor without a history.  but:
```
with torch.no_grad():
    return x
```
will return a tensor with history

For 3) This happens rarely, only if you explicitly don't want to share the version counter (e.g. because autograd would complain about the change but you happen to know it is okay, or the version counter is being explicitly tested).

The expected steps are:
  - [ ] `test/test_autograd.py` (except for the version counter test)

cc @ezyang @SsnL @albanD @zou3519 @gqchen
